onnxruntime/test/onnx/TestCase.cc

```diff
@@ -52,100 +52,111 @@
-Status Convert(const google::protobuf::Map<std::string, int64_t>& data, MapStringToInt64** out) {
-  ConvertMap(data, out);
-  return Status::OK();
+ONNXTensorElementDataType NumericTypeToONNXType<float>() {
+  return ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT;
 }
 
 template <>
-Status Convert(const google::protobuf::Map<std::string, float>& data, MapStringToFloat** out) {
-  ConvertMap(data, out);
-  return Status::OK();
+ONNXTensorElementDataType NumericTypeToONNXType<double>() {
+  return ONNX_TENSOR_ELEMENT_DATA_TYPE_DOUBLE;
 }
 
 template <>
-Status Convert(const google::protobuf::Map<std::string, double>& data, MapStringToDouble** out) {
-  ConvertMap(data, out);
-  return Status::OK();
+ONNXTensorElementDataType NumericTypeToONNXType<int64_t>() {
+  return ONNX_TENSOR_ELEMENT_DATA_TYPE_INT64;
 }
 
 template <>
-Status Convert(const google::protobuf::Map<int64_t, std::string>& data, MapInt64ToString** out) {
-  ConvertMap(data, out);
-  return Status::OK();
+ONNXTensorElementDataType NumericTypeToONNXType<std::string>() {
+  return ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING;
 }
 
-template <>
-Status Convert(const google::protobuf::Map<int64_t, int64_t>& data, MapInt64ToInt64** out) {
-  ConvertMap(data, out);
-  return Status::OK();
+template <typename T>
+OrtValue* CreateTensorWithDataAsOrtValue(OrtAllocatorInfo* info, std::vector<T>& input) {
+  std::vector<int64_t> dims(1, input.size());
+  OrtValue* ret = nullptr;
+  ORT_THROW_ON_ERROR(::OrtCreateTensorWithDataAsOrtValue(info, input.data(), input.size() * sizeof(T), dims.data(),
+                                                         dims.size(), NumericTypeToONNXType<T>(), &ret));
+  return ret;
 }
 
-template <>
-Status Convert(const google::protobuf::Map<int64_t, float>& data, MapInt64ToFloat** out) {
-  ConvertMap(data, out);
-  return Status::OK();
+template <typename key_type, typename value_type>
+OrtValue* PbMapToMLValue(const google::protobuf::Map<key_type, value_type>& map) {
+  OrtAllocatorInfo* info;
+  ORT_THROW_ON_ERROR(OrtCreateAllocatorInfo("Cpu", OrtDeviceAllocator, 0, OrtMemTypeDefault, &info));
+  std::unique_ptr<OrtAllocatorInfo, decltype(&OrtReleaseAllocatorInfo)> rel_info(info, OrtReleaseAllocatorInfo);
+  const size_t ele_count = map.size();
+  std::vector<int64_t> dims(1, ele_count);
+  std::vector<key_type> keys(ele_count);
+  std::vector<value_type> values(ele_count);
+  size_t i = 0;
+  for (auto& kvp : map) {
+    keys[i] = kvp.first;
+    values[i] = kvp.second;
+    ++i;
+  }
+  std::vector<OrtValue*> map_in(2);
+  map_in[0] = CreateTensorWithDataAsOrtValue(info, keys);
+  if (map_in[0] == nullptr) ORT_THROW("Create keys tensor failed");
+
+  map_in[1] = CreateTensorWithDataAsOrtValue(info, values);
+  if (map_in[1] == nullptr) ORT_THROW("Create values tensor failed");
+
+  // create map ort value
+  OrtValue* map_ort = nullptr;
+  ORT_THROW_ON_ERROR(OrtCreateValue(map_in.data(), 2, ONNX_TYPE_MAP, &map_ort));
+  for (auto value : map_in) OrtReleaseValue(value);
```

Comment:
can map_in be arranged to use RAII?

Reply:
This is test code, if it has memory leak because of unexpected error, I think it doesn't matter.
RAII will make the code more complex, because it needs two contains, one for raw pointer, another for RAII.

Reply:
Thanks. Will fix it.